modules/security-understanding-container-validation.adoc

```diff
@@ -0,0 +1,157 @@
+// Module included in the following assemblies:
+//
+// * security/container_security/security-understanding.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="security-understanding-container-validation_{context}"]
+= Container security and validation
+
+{product-title} is released as a set of container images that comprise the different functions of the software.
+These images must be validated in order to ensure that they have not been tampered with by an external party, and to ensure that they are the intended images provided by Red{nbsp}Hat.
+
+[id="security-understanding-image-integrity_{context}"]
+== Image integrity validation at pull time
+
+When images are downloaded, or pulled, from a registry, an {product-title} cluster performs the following actions to ensure the integrity of each image:
+
+. The cluster downloads an image manifest and calculates its digest.
+
+. The cluster compares the calculated digest of the image manifest with the known digest.
+If the digest values do not match, the image is rejected without further processing.
+
+. The cluster downloads an image config file, which contains metadata such as SHA256 digests for each part of the container image, and calculates the digest of the image config file.
+
+. The cluster compares the calculated digest of the image config file with the digest that is provided in the image manifest.
+If the digest values do not match, the image is rejected without further processing.
+
+. For each file that comprises the container image, the cluster downloads the file and calculates its digest.
+
+. The cluster compares the calculated digest of each file that makes up the image with the corresponding digest that is provided in the image manifest.
+If the digest values do not match, the image is rejected.
+
+Once the cluster completes all of the previous steps, the integrity of the image has been validated.
+
+[NOTE]
+====
+The image manifest is validated only when the manifest digest is known, which is true in the following cases:
+
+. The image is pulled by specifying the SHA256 image digest, for example `quay.io/openshift-release-dev/ocp-release@sha256:ea7ac3ad42169b39fce07e5e53403a028644810bee9a212e7456074894df40f3`.
+
+. The image signature is also being validated. See “Image authenticity validation” for more information.
+====
+
+[id="security-understanding-image-authenticity_{context}"]
+== Image authenticity validation
+
+In addition to validating the integrity of images, {product-title} clusters also validate their authenticity by certifying that the images were approved by a trusted source, in this case Red{nbsp}Hat.
```

Comment:
> In the context of OpenShift installs/updates, right now, this version binding IIRC not used...

Correct.  But the only images Red Hat is currently signing under `quay.io/openshift-release-dev/ocp-release` are OCP release images.  And when a version string is part of the update request, divergence between the requested version and the actual release image version is fatal since 2020's openshift/cluster-version-operator#431.  So the "is the requested image the OCP release image you're hoping for?" are happening today, just not at the pullspec-asserting-signature level.

Reply:
The original word used was "published", though @wking suggested to change it to "approved" with the following reasoning given:

> The signer and publisher can be different, e.g. a customer's ProdSec team might sign some subset of Red Hat release images as acceptable, and then those releases would have both Red Hat and customer-ProdSec signatures

So I'm not sure where to proceed if "approved" is too ambiguous. Maybe some rewording of that paragraph to convey the point about "was produced by a RH build system and not modified since"? Maybe something like this:
```suggestion
In addition to validating the integrity of images, {product-title} clusters also validate their authenticity by certifying that the images have not been modified since being produced by a trusted source, in this case Red{nbsp}Hat.
```

Reply:
I'm going to step back and suggest that:

> In addition to validating the integrity of images, {product-title} clusters also validate their authenticity by certifying that the images were approved by a trusted source, in this case Red{nbsp}Hat.
> This validation process includes checking the digital signature of the images.
>
> Cluster administrators can use `ClusterImagePolicy` resources to declare image policy for the cluster.

get reworded to something like:

> Cluster administrators can use `ClusterImagePolicy` resources to declare image policy for the cluster, requiring signature validation and other policy checks in addition to image integrity.

Although other wording in the current pull is a bit mushy on whether "integrity" includes policy, or is limited to the internally-consistent digest-matching and excluding higher-level policy like sufficient signatures.

Reply:
Okay, I'll go with your suggested wording for now, and if need be it can be further refined here or in a v2 of this doc.